### DISCUSSION TITLE
How to get peak by cell matrix from ChromatinAssay class of Seurat object?

### DISCUSSION THREAD

**Discussion by ananyapavuluri** [stuart-lab/signac#1373] [Q&A] [answered]
Hello, I am brand new to Seurat and Signac. I have a large Seurat object (S4 object of class Seurat) representing an scATAC-seq experiment. Under assays, I have `ATAC ` which is an S4 object of class `ChromatinAssay`, and under this, there is `counts` which is of class `dgCMatrix`. 

I am wondering how I can extract the peak by cell matrix from `counts`, and am also wondering what `i `and `p` represent under `counts`, because`i` is not matching the number of peaks that should be in the matrix.

**Comment by timoast** [comment, marked as answer]
Hi, you can use the `GetAssayData()` function

The `i`, `p` slots are part of the internal data representation for a sparse matrix. Please do not interact with those slots directly
